### ISSUE TITLE
Strange caching behaviour on the checkout level

### ISSUE THREAD

**Issue by korycins** [mirumee/saleor-storefront#853]
### Steps to reproduce the problem
1. Log in 
2. add to cart
3. go to checkout
You will be moved to the 2nd step of checkout - shipping-method
![Aug-11-2020 12-23-21](https://user-images.githubusercontent.com/23282005/89886815-86871600-dbcd-11ea-8e29-e7c7a0f85ff9.gif)

**System information**
Operating system:  OSX
Browser: Chrome
Storefront Branch: master

**Comment by orzechdev** [comment]
What do you mean by strange caching behaviour? And what is the expected result?

**Comment by korycins** [comment]
I think that you should watch the gif that I've uploaded.

**Comment by orzechdev** [comment]
Ok, do you mean that the user should be moved to the first step instead of the second?

**Comment by korycins** [comment]
Based on the gif. 
- Do you see how many shipping methods are there? 
- user should be able to pick up own address before  we will show the shipping methods. Currently list is empty as I didn't have a possibility to pick up  my shipping or billing address.

**Comment by orzechdev** [comment]
Ok, thank you for the issue submission.

I have tested it thoroughly and I see a few things:
- I wouldn't reproduce the problem as after adding an item to the cart and proceeding to the checkout I am going to the first step, not the second. Moreover, we have covered adding an item to the cart and buying it in e2e tests, which pass.
- I guess it is possible when a user first adds something to the cart, sets shipping and billing addresses, then go to cart and remove a previously added item to the cart, log out (we are sure that cache and local storage is cleared now), log in again, add an item to the cart, go to the checkout and now indeed the user is redirected to the second step. I have checked additionally the network request and indeed user has set a checkout object already, which already has addresses set, from what we can infer we should redirect the user to the second step, but it hasn't shipping methods determined:
![Zrzut ekranu 2020-08-19 o 11 40 51](https://user-images.githubusercontent.com/9825562/90619412-80b6b380-e211-11ea-8d1c-d21d30b0bc79.png)
After second item addition to the cart although we have shipping methods updated:
![Zrzut ekranu 2020-08-19 o 11 55 25](https://user-images.githubusercontent.com/9825562/90620571-ee171400-e212-11ea-9958-43c33bebc384.png)

What I see here, the problem is most probably with not showing updated shipping methods (after adding item to the cart for the second time and visiting checkout page the second time). Could such a scenario of cases also occur in your situation @korycins ?
